src/account/UpgradeableModularAccount.sol

```diff
@@ -136,7 +135,7 @@ contract UpgradeableModularAccount is
             revert UnrecognizedFunction(msg.sig);
         }
 
-        _checkPermittedCallerIfNotFromEP();
+        _checkPermittedCallerAndAssociatedHooks();
```

Comment:
What if a signer of the account call an execFunction? Would it fail [here](https://github.com/erc6900/reference-implementation/pull/90/files#diff-4218469ba4b89c8a36e355bbf535a3d32fbe4f6846506a6dc961b5ec5cd2f1d7R645)?

Reply:
I.e. if a signer, as set in the storage of `SingleSignerValidation`, tries to call the account? In that case, this check would fail, the signer would need to use `executeWithAuthorization` to trigger runtime validation.

Alternatively, if an account is only used from an EOA, you could add an EOA itself as an allowed caller by installing the EOA address + the direct call magic value as a validation. But, this wouldn't be usable via user op validation.

Reply:
Aren't we executing the pre-execution hooks currently twice in case a permitted caller is calling through fallback? Within `_checkPermittedCallerAndAssociatedHooks()` in `L664` we do the pre-execution hooks of the function selector, but again in the following `fallback()` flow in `L143`.
Also, are we not forgetting to do the post-permission hooks in the fallback flow? 👀

Reply:
@0xrubes you are correct... we'll get a fix PR up soon.